<a href="https://colab.research.google.com/github/rahulsait/ExcelR-codes/blob/main/MODEL_VALIDATION_METHODS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Train and Test Set :**

In [1]:
# import T-T libs :

from pandas import read_csv
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

In [2]:
# load the dataset :

filename = "pima-indians-diabetes.data.csv"
names = ["preg","plas","pres","skin","test","mass","pedi","age","class"]
dataframe = read_csv(filename, names = names)
array = dataframe.values

X = array[:,0:8]
Y = array[:,8]
test_size = 0.33
seed = 7

In [4]:
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = test_size, random_state = seed)
model = LogisticRegression(max_iter = 500)
model.fit(X_train, Y_train)
result = model.score(X_test, Y_test)

In [5]:
result*100.00

78.74015748031496

### **Accuracy of the Model is 78.74**

# **K-Fold-Cross-Validation :**

### **K = 10**

In [1]:
# import KFCV lib :

from pandas import read_csv
from sklearn.model_selection import KFold
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression

In [3]:
# load the dataset :

filename = "pima-indians-diabetes.data.csv"
names = ["preg","plas","pres","skin","test","mass","pedi","age","class"]
dataframe = read_csv(filename, names = names)
array = dataframe.values

X = array[:,0:8]
Y = array[:,8]

In [4]:
num_folds = 10
seed = 7

In [7]:
# create an object to build the Model :

kfold = KFold(n_splits = num_folds)
model = LogisticRegression(max_iter = 400)
results = cross_val_score(model, X, Y, cv = kfold)
results

array([0.68831169, 0.81818182, 0.76623377, 0.68831169, 0.77922078,
       0.79220779, 0.84415584, 0.83116883, 0.75      , 0.80263158])

In [6]:
results.mean()*100

77.60423786739577

In [8]:
results.std()*100 # std < 10 : datapoints are very close to eachother

5.157545262086822

### **Accuracy of the Model is 77.60**

# **Leave-One-Out-Cross-Validation :**

In [9]:
# import LOOCV lib :

from pandas import read_csv
from sklearn.model_selection import LeaveOneOut
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression

In [10]:
# load the dataset :

filename = "pima-indians-diabetes.data.csv"
names = ["preg","plas","pres","skin","test","mass","pedi","age","class"]
dataframe = read_csv(filename, names = names)
array = dataframe.values

X = array[:,0:8]
Y = array[:,8]

In [11]:
# create an object to build the Model :

loocv = LeaveOneOut()
model = LogisticRegression(max_iter = 300)
results = cross_val_score(model, X, Y, cv = loocv)

In [12]:
results.mean()*100

77.60416666666666

### **Accuracy of the Model is 77.60**